crates/spawner/src/destroyer.rs

```diff
@@ -27,7 +63,28 @@ fn destroy(
             if let ObjectType::Active(active_type) = object_type {
                 counter.player_mut(player).unwrap().update(active_type, -1);
             }
-            commands.entity(entity).despawn_recursive();
+            commands.entity(entity).insert(ToBeDespawned);
```

Comment:
I won't do this via marker trait `ToBeDespawned` but via an event to avoid the need to wait until the next base (for commands being flushed to World).

What we can do:

* keep `despawn_recursive()` here (the entity stays available until next stage)
* send `Despavning(Entity)` event from here
* register `send_data<T>` via a public plugin `DespawnEventsPlugin<T>` so that it runs in Update stage (while the entities are still available) but `.after()` `find_dead`
* From everywhere it is needed, one can add the appropriate `DespawnEventsPlugin<T>`

What do you think?